// @fluentui/utilities/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMergedRef = exports.setSSR = exports.IsFocusVisibleClassName = exports.setFocusVisibility = exports.omit = exports.values = exports.shallowCompare = exports.mapEnumByName = exports.filteredAssign = exports.assign = void 0;
var tslib_1 = require("tslib");
tslib_1.__exportStar(require("./Async"), exports);
tslib_1.__exportStar(require("./AutoScroll"), exports);
tslib_1.__exportStar(require("./BaseComponent"), exports);
tslib_1.__exportStar(require("./BaseComponent.types"), exports);
tslib_1.__exportStar(require("./DelayedRender"), exports);
tslib_1.__exportStar(require("./EventGroup"), exports);
tslib_1.__exportStar(require("./FabricPerformance"), exports);
tslib_1.__exportStar(require("./GlobalSettings"), exports);
tslib_1.__exportStar(require("./IClassNames"), exports);
tslib_1.__exportStar(require("./IComponentAs"), exports);
tslib_1.__exportStar(require("./IDisposable"), exports);
tslib_1.__exportStar(require("./Point"), exports);
tslib_1.__exportStar(require("./IRectangle"), exports);
tslib_1.__exportStar(require("./IRenderComponent"), exports);
tslib_1.__exportStar(require("./IRenderFunction"), exports);
tslib_1.__exportStar(require("./ISize"), exports);
tslib_1.__exportStar(require("./IStyleFunction"), exports);
tslib_1.__exportStar(require("./KeyCodes"), exports);
tslib_1.__exportStar(require("./Rectangle"), exports);
tslib_1.__exportStar(require("./appendFunction"), exports);
tslib_1.__exportStar(require("./aria"), exports);
tslib_1.__exportStar(require("./array"), exports);
tslib_1.__exportStar(require("./asAsync"), exports);
tslib_1.__exportStar(require("./assertNever"), exports);
tslib_1.__exportStar(require("./classNamesFunction"), exports);
tslib_1.__exportStar(require("./componentAs/composeComponentAs"), exports);
tslib_1.__exportStar(require("./controlled"), exports);
tslib_1.__exportStar(require("./createRef"), exports);
tslib_1.__exportStar(require("./css"), exports);
tslib_1.__exportStar(require("./customizations/Customizations"), exports);
tslib_1.__exportStar(require("./customizations/Customizer"), exports);
tslib_1.__exportStar(require("./customizations/Customizer.types"), exports);
tslib_1.__exportStar(require("./customizations/CustomizerContext"), exports);
tslib_1.__exportStar(require("./customizations/customizable"), exports);
tslib_1.__exportStar(require("./customizations/useCustomizationSettings"), exports);
tslib_1.__exportStar(require("./customizations/mergeCustomizations"), exports);
tslib_1.__exportStar(require("./customizations/mergeSettings"), exports);
tslib_1.__exportStar(require("./dom"), exports);
tslib_1.__exportStar(require("./extendComponent"), exports);
tslib_1.__exportStar(require("./focus"), exports);
tslib_1.__exportStar(require("./getId"), exports);
tslib_1.__exportStar(require("./getNativeElementProps"), exports);
tslib_1.__exportStar(require("./hoist"), exports);
tslib_1.__exportStar(require("./hoistStatics"), exports);
tslib_1.__exportStar(require("./initializeComponentRef"), exports);
tslib_1.__exportStar(require("./initializeFocusRects"), exports);
tslib_1.__exportStar(require("./useFocusRects"), exports);
tslib_1.__exportStar(require("./initials"), exports);
tslib_1.__exportStar(require("./keyboard"), exports);
tslib_1.__exportStar(require("./language"), exports);
tslib_1.__exportStar(require("./math"), exports);
tslib_1.__exportStar(require("./memoize"), exports);
tslib_1.__exportStar(require("./merge"), exports);
tslib_1.__exportStar(require("./mobileDetector"), exports);
tslib_1.__exportStar(require("./modalize"), exports);
var object_1 = require("./object");
Object.defineProperty(exports, "assign", { enumerable: true, get: function () { return object_1.assign; } });
Object.defineProperty(exports, "filteredAssign", { enumerable: true, get: function () { return object_1.filteredAssign; } });
Object.defineProperty(exports, "mapEnumByName", { enumerable: true, get: function () { return object_1.mapEnumByName; } });
Object.defineProperty(exports, "shallowCompare", { enumerable: true, get: function () { return object_1.shallowCompare; } });
Object.defineProperty(exports, "values", { enumerable: true, get: function () { return object_1.values; } });
Object.defineProperty(exports, "omit", { enumerable: true, get: function () { return object_1.omit; } });
tslib_1.__exportStar(require("./osDetector"), exports);
tslib_1.__exportStar(require("./overflow"), exports);
tslib_1.__exportStar(require("./properties"), exports);
tslib_1.__exportStar(require("./renderFunction/composeRenderFunction"), exports);
tslib_1.__exportStar(require("./resources"), exports);
tslib_1.__exportStar(require("./rtl"), exports);
tslib_1.__exportStar(require("./safeRequestAnimationFrame"), exports);
tslib_1.__exportStar(require("./safeSetTimeout"), exports);
tslib_1.__exportStar(require("./scroll"), exports);
tslib_1.__exportStar(require("./selection/index"), exports);
tslib_1.__exportStar(require("./string"), exports);
tslib_1.__exportStar(require("./styled"), exports);
tslib_1.__exportStar(require("./warn"), exports);
tslib_1.__exportStar(require("./ie11Detector"), exports);
tslib_1.__exportStar(require("./getPropsWithDefaults"), exports);
var setFocusVisibility_1 = require("./setFocusVisibility");
Object.defineProperty(exports, "setFocusVisibility", { enumerable: true, get: function () { return setFocusVisibility_1.setFocusVisibility; } });
Object.defineProperty(exports, "IsFocusVisibleClassName", { enumerable: true, get: function () { return setFocusVisibility_1.IsFocusVisibleClassName; } });
var setSSR_1 = require("./dom/setSSR");
Object.defineProperty(exports, "setSSR", { enumerable: true, get: function () { return setSSR_1.setSSR; } });
var createMergedRef_1 = require("./createMergedRef");
Object.defineProperty(exports, "createMergedRef", { enumerable: true, get: function () { return createMergedRef_1.createMergedRef; } });
require("./version");
